### ISSUE TITLE
Mysql CREATE TABLE parse fails on COLLATE/CHARSET keyword without equal sign

### ISSUE THREAD

**Issue by jblotus** [sqlfluff/sqlfluff#3664]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I am trying to lint some create table statements and got parsing errors when using the `COLLATE` keyword if that keyword was not followed with an equal sign.

I also tested this with the `CHARSET` keyword and I had the same parse issue.

### Expected Behaviour

I expect the parsing to work with or without the equal sign since this is legal in mysql.

### Observed Behaviour

`CREATE TABLE` statements with `CHARSET` or `COLLATE` without a trailing equals sign fail to parse.
This might apply to other constructs that are part of a `CREATE TABLE` statement.

### How to reproduce

try to lint the following using mysql dialect:

```sql

## this works
CREATE TABLE works_with_equals (
    id INT
) COLLATE=utf8_general_ci;

## this fails to parse
CREATE TABLE fails_no_equals (
    id INT
) COLLATE utf8_general_ci;

```

### Dialect

dialect: mysql

### Version

version 1.2.1

### Configuration

stock configuration

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jblotus** [comment]
This is the only possibly related issue I could find: https://github.com/sqlfluff/sqlfluff/issues/814

**Comment by tunetheweb** [comment]
I think this should be as easy as  changing `Ref("EqualsSegment"),` to `Ref("EqualsSegment", optional=True)` here:

https://github.com/sqlfluff/sqlfluff/blob/7a40bf6e4dad0d5f80523e077c8a20c0aa9bc987/src/sqlfluff/dialects/dialect_mysql.py#L292-L297

And adding a test case case.
